modules/_labs/side-panel/react/README.md

```diff
@@ -0,0 +1,131 @@
+</a>  This component is work in progress and currently in pre-release.
+
+A collapsable side panel
+
+## Installation
+
+```sh
+yarn add @workday/canvas-kit-labs-react-side-panel
+```
+
+## Usage
+
+```tsx
+import SidePanel, {useSidePanel} from '@workday/canvas-kit-labs-react-side-panel';
+
+/**
+ * A SidePanel is made up of three components
+ * - The panel itself <SidePanel>
+ * - An accessible name (either an existing element or you can use <span hidden>)
+ * - A control that expands/collapses the SidePanel, usually <SidePanel.ToggleButton> as a child of <SidePanel>
+ * A convenience hook is available for setting the state of the SidePanel along with the proper aria- attributes
+ *
+ */
+const [panelProps, labelProps, controlProps] = useSidePanel({});
+
+<SidePanel {...panelProps}>
+  <SidePanel.ToggleButton {...controlProps} />
+  <h3 {...labelProps}>Tasks Panel</h3>
+  {/* Your panel contents */}
+</SidePanel>
+
+```
+## Hooks
+### `useSidePanel`
+This hook manages the state and `aria-` attributes for the SidePanel. It takes in a `config` object:
+* `intialExpanded: boolean`: Specifies the initial expanded/collapsed state of the side panel.
+* `panelId: string`: specifies the id of the panel, if `undefined` then this hook will generate one
+* `labelId: string` specifies the id of the label, if `undefined` then this hook will generate one
+
+and returns:
+* `expanded: boolean`: The state for the `SidePanel`'s `expanded` prop.
+* `setExpanded: (expanded?: boolean) => void`: A function that sets the `expanded` state.
+* `panelProps: object`: Contains the `expanded`, `id`, and `aria-labelledby` props for the `SidePanel` (or equivalent) component.
+* `labelProps: object`: Contains the `id` prop for the label.
+* `controlProps: object`: Contains the `aria-controls`, `aria-expanded`, `aria-labelledby`, and click handler for the side panel's control element.
+
+#### Usage
+```tsx
+const [expanded, setExpanded, panelProps, labelProps, controlProps] = useSidePanel({
+  initialExpanded: true,
+  panelId: 'side-panel-1',
+  labelId: 'side-panel-label-1'
+});
+
+<SidePanel {...panelProps}>
+  {/* Make sure the control is first focusable */}
+  <SidePanel.ToggleButton {...controlProps} />
+  <span hidden {...labelProps}>Example Panel</span>
+  {/* Rest of the children here */}
+</SidePanel>
+
+or
+
+<SidePanel {...panelProps}>
+  {/* If you want to use your own button */}
+  <CustomControl {...controlProps} />
+  {/* If you already have an element that can be an accessible name for the panel */}
+  <h3 {...labelProps}>Example Panel</h3>
+</SidePanel>
+```
+
+## Static Properties
+
+#### `ToggleButton`
+> `<SidePanel.ToggleButton>` is a control that is meant to toggle between `expanded = true` and `expanded = false` states. It must be used within the `SidePanel` component as a child. Use in conjunction with `useSidePanel`'s `controlProps`, otherwise it does not come with explicit `onClick` handlers.
```

Comment:
Yea it will still take `aria-label`, it's just optional now. Honestly, we should change the base `IconButton` to take either `aria-label` or `aria-labelledby` anyway.

Reply:
Tracked here: https://github.com/Workday/canvas-kit/issues/894